src/wallet/wallet.ts

```diff
@@ -122,18 +136,130 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
+  clearSensitiveData() {
+    this.seed = null;
+    this.authPrivKey = null;
+  }
+
+  /**
+   * Get auth xpubkey in the 280' purpose path from seed
+   *
+   * @param {String} seed 24 words
+   * @param {Object} options Options with passphrase and networkName
+   *
+   * @return {String} auth xpubkey
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  static getAuthXPubKeyFromSeed(seed: string, options: { passphrase?: string, networkName?: string } = {}): string {
+    const methodOptions = Object.assign({
+      passphrase: '',
+      networkName: 'mainnet',
+    }, options);
+
+    const xpriv = walletUtils.getXPrivKeyFromSeed(seed, methodOptions);
+    const privkey = HathorWalletServiceWallet.deriveAuthPrivateKey(xpriv);
+
+    return privkey.xpubkey;
+  }
+
+  /**
+   * Derive private key from root to the auth specific purpose derivation path
+   *
+   * @param {HDPrivateKey} xpriv The wallet's root xpriv
+   *
+   * @return {HDPrivateKey} Derived private key at the auth derivation path
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  static deriveAuthPrivateKey(xpriv: bitcore.HDPrivateKey): bitcore.HDPrivateKey {
+    return xpriv.deriveNonCompliantChild(WALLET_SERVICE_AUTH_DERIVATION_PATH);
+  }
+
+  /**
+   * Encrypt and store wallet data on storage using the wallet's PIN code
+   *
+   * @param {string} pin Pin code to use as password to encrypt the auth key
+   * @param {string} mainKey xpriv of the change level derivation to sign inputs
+   * @param {string} authKey xpriv of the auth specific purpose derivation path
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  static initializeAccessData(pin: string, mainKey: string, authKey: string) {
+    const initialAccessData = wallet.getWalletAccessData() || {};
+
+    const encryptedAuthKey = wallet.encryptData(authKey, pin);
+    const encryptedMainKey = wallet.encryptData(mainKey, pin);
+
+    initialAccessData['authKey'] = encryptedAuthKey.encrypted.toString();
+    initialAccessData['mainKey'] = encryptedMainKey.encrypted.toString();
+
+    wallet.setWalletAccessData(initialAccessData);
+  }
+
   /**
    * Start wallet: load the wallet data, update state and start polling wallet status until it's ready
    *
+   * @param {Object} optionsParams Options parameters
+   *  {
+   *   'pinCode': PIN to encrypt the auth xpriv on storage
+   *  }
+   *
    * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async start() {
+  async start({ pinCode }) {
+    if (!this.seed) {
+      throw new Error('Seed should be in memory when starting the wallet.');
+    }
+
+    if (!pinCode) {
+      throw new Error('Pin code is required when starting the wallet.');
+    }
+
     this.setState(walletState.LOADING);
+
+    const xpriv = walletUtils.getXPrivKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
     const xpub = walletUtils.getXPubKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
```

Comment:
This is interesting

`getXPrivKeyFromSeed` will return the root xprivkey
`getXPubKeyFromSeed` will return a xpubkey at the account derivation path (which is the one we use on the wallet service)

So I can't do `xpriv.xpubkey` directly

Maybe we should refactor the method's name to something like `getAccountPathXPubKeyFromSeed`, but that would be a breaking change

What do you think?

Reply:
Funny thing! Do you know why doesn't `getXPrivKeyFromSeed` derivate at the account level as well? We might create `getAccountPathXPubKeyFromSeed` and deprecate `getXPubKeyFromSeed`. But the variable names are misleading because xpriv and xpub are at different derivation levels.

Reply:
We actually need `getXPrivKeyFromSeed` as it is on many parts of the lib, including `getXPubKeyFromSeed`

I guess the question is why `getXPubKeyFromSeed` is named this way. Maybe this was done to maintain compatibility?

@pedroferreira1 can you help?